woocommerce/wc-admin
# Components: Add SearchListControl

Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/417 – port over the SearchListComponent, which is a base component used to create all the searchable + selectable lists in our blocks. This PR adds the components, doc file, and a devdocs entry.

I also fixed the components changelog, since it somehow flipped back to saying 1.5.0 was unreleased.

### Accessibility




### Screenshots

The block on its own doesn't handle any data requests, it expects a list of data, and a subset of selected data.

Example used in a block, populated with category content.
![block](https://user-images.githubusercontent.com/541093/53053511-8fcb7280-346f-11e9-89c4-08ea047cf373.png)

And the basic example in devdocs:
![devdocs](https://user-images.githubusercontent.com/541093/53053510-8fcb7280-346f-11e9-9681-a11580406414.png)

### Detailed test instructions:

- Check out the component in devdocs: `/wp-admin/admin.php?page=wc-admin#/devdocs/search-list-control`

To test "in action" in the blocks project:

- On this branch, after building, move to the components directory
- Run `npm link` to set up a symlink for the @woocommerce/components package
- Grab this branch of the woo blocks: https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/442
- In the blocks project folder, run `npm link @woocommerce/components` to point this project's dependency to the linked package
- Build the blocks project, `npm run build` (note, **don't** run `npm install` between the link and build commands, it'll blow away your link)
- Add "Products by Category", "Products by Attribute", "Hand-picked Products", or a "Featured Product" block to your post, all use controls based on the `SearchListControl`.
- Expect: these should work as usual, no visual or functional changes.
